code/datums/mind.dm

```diff
@@ -226,11 +224,7 @@
 	for(var/datum/antagonist/A as anything in antag_datums)
 		if(A.has_antag_objectives(include_team)) // this checks teams also
 			return TRUE
-	// For custom non-antag role teams
-	if(include_team && LAZYLEN(teams))
-		for(var/datum/team/team as anything in teams)
-			if(team.objective_holder.has_objectives())
-				return TRUE
```

Comment:
I dont think this fixes the issue that teams variable was created for. Essentially, if a mind is added to a team, with no antag_datum assigned to it, their mind has no connection to the team. When they request to look at their objectives without an antag_datum given by the team, they will not be able to see their teams objectives. This is why this was originally included. I think removing this is going to cause more issues. I'm sure theres a good fix, but I cannot think of one

Reply:
I think we need to move generic teams back, and create subtype for antag teams and include required functionality, related to antag datums, there

Reply:
And, I don't really see reasons to separately search for objectives in antag datum teams, we just need to make sure, that all teams are added to 'team' list, and search for objectives there